### ISSUE TITLE
x64dbg two fatal bug

### ISSUE THREAD

**Issue by number201724** [x64dbg/x64dbg#1960]
Bug 1:
If the EXE EntryPoint is 0, Entry Breakpoint breaks to the PE Header's IMAGE_DOS_SIGNATURE

Bug2:
If you use ZwMapViewOfSection to load 2 copies of ntdll.dll (path is consistent) then the breakpoint on this module will be wrong. ZwOpenFile -> ZwCreateSection -> ZwMapViewOfSection

**Comment by number201724** [comment]
Some programs use TLS CALLBACK to start.

**Comment by mrexodia** [comment]
Zero is a valid entry point. See https://corkamiwiki.github.io/PE

For bug 2, could you provide some clearer information? Example executable or steps perhaps.

**Comment by number201724** [comment]
Yes 0 is a valid entry point, but the breakpoint of x64dbg is in the PE header instead of the code segment.

**Comment by number201724** [comment]
Bug2 I will write a demo later

**Comment by abbasfull** [comment]
No problem

**Comment by number201724** [comment]
bug example:
https://github.com/number201724/x64dbg_bugreport

**Comment by number201724** [comment]
Example 1: When the entry breakpoint is turned on, he crashes in ntdll

**Comment by number201724** [comment]
For 0 entry point, he should not set the breakpoint to the IMAGE_DOS_SIGNATURE of the PE header.

**Comment by number201724** [comment]
This will cause the ntdll LdrXXXXX function to crash.

**Comment by Mattiwatti** [comment]
Alright alright. Your first PE now runs for me with the above PR applied.

**Comment by number201724** [comment]
After a few months, there is still no repair

**Comment by torusrxxx** [comment]
I support the idea to add number suffix to duplicated DLL.

**Comment by Mattiwatti** [comment]
The problem with adding the image base to the module name (unless this name is only used for display/GUI purposes) is that it would break any user breakpoints and patches upon the next start (due to ASLR), no? That said, for display purposes having the base address appended does seem useful.

If we make the PE malicious enough this seems like a problem that is not solvable in the general case IMO:
1. Genuine `ntdll.dll` gets loaded
2. Process loads a pseudorandom number (say, between 10 and 30) of `ntdll.dll`s based on some internal algorithm and maps them as `SEC_IMAGE`. Note that you can specify a preferred load address, which, assuming the requested address range is not already taken, will work. This 'ASLR-ness' could also be randomized for each module, making some load at the requested base address, and others at a random one.
3. The process now has `X` versions of `ntdll.dll` to choose from (and also to compare memory between to check for hooks/patches).

In the face of a PE that does something like the above, do you see any possible way to store the relocation and patch info to the database in a way that is still valid after restarting the executable?
